File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-eq-shared-5-4/StubCalculationPeriod.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class StubCalculationPeriod : ISerialized
    {
        public StubCalculationPeriod(XmlNode xmlNode)
        {
            XmlNode initialStubNode = xmlNode.SelectSingleNode("initialStub");
            
            if (initialStubNode != null)
            {
                if (initialStubNode.Attributes["href"] != null || initialStubNode.Attributes["id"] != null) 
                {
                    if (initialStubNode.Attributes["id"] != null) 
                    {
                        initialStubIDRef_ = initialStubNode.Attributes["id"].Value;
                        Stub ob = new Stub(initialStubNode);
                        IDManager.SetID(initialStubIDRef_, ob);
                    }
                    else if (initialStubNode.Attributes["href"] != null)
                    {
                        initialStubIDRef_ = initialStubNode.Attributes["href"].Value;
                    }
                    else
                    {
                        initialStub_ = new Stub(initialStubNode);
                    }
                }
                else
                {
                    initialStub_ = new Stub(initialStubNode);
                }
            }
            
        
            XmlNode finalStubNode = xmlNode.SelectSingleNode("finalStub");
            
            if (finalStubNode != null)
            {
                if (finalStubNode.Attributes["href"] != null || finalStubNode.Attributes["id"] != null) 
                {
                    if (finalStubNode.Attributes["id"] != null) 
                    {
                        finalStubIDRef_ = finalStubNode.Attributes["id"].Value;
                        Stub ob = new Stub(finalStubNode);
                        IDManager.SetID(finalStubIDRef_, ob);
                    }
                    else if (finalStubNode.Attributes["href"] != null)
                    {
                        finalStubIDRef_ = finalStubNode.Attributes["href"].Value;
                    }
                    else
                    {
                        finalStub_ = new Stub(finalStubNode);
                    }
                }
                else
                {
                    finalStub_ = new Stub(finalStubNode);
                }
            }
            
        
        }
        
    
        #region InitialStub_
        private Stub initialStub_;
        public Stub InitialStub_
        {
            get
            {
                if (this.initialStub_ != null)
                {
                    return this.initialStub_; 
                }
                else if (this.initialStubIDRef_ != null)
                {
                    initialStub_ = IDManager.getID(initialStubIDRef_) as Stub;
                    return this.initialStub_; 
                }
                else
                {
                      return this.initialStub_; 
                }
            }
            set
            {
                if (this.initialStub_ != value)
                {
                    this.initialStub_ = value;
                }
            }
        }
        #endregion
        
        public string initialStubIDRef_ { get; set; }
        #region FinalStub_
        private Stub finalStub_;
        public Stub FinalStub_
        {
            get
            {
                if (this.finalStub_ != null)
                {
                    return this.finalStub_; 
                }
                else if (this.finalStubIDRef_ != null)
                {
                    finalStub_ = IDManager.getID(finalStubIDRef_) as Stub;
                    return this.finalStub_; 
                }
                else
                {
                      return this.finalStub_; 
                }
            }
            set
            {
                if (this.finalStub_ != value)
                {
                    this.finalStub_ = value;
                }
            }
        }
        #endregion
        
        public string finalStubIDRef_ { get; set; }
        public string choiceStr_0;
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-eq-shared-5-4/StubCalculationPeriod.cpp
// StubCalculationPeriod.cpp 
#include "StubCalculationPeriod.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

StubCalculationPeriod::StubCalculationPeriod(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //initialStubNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* initialStubNode = xmlNode->FirstChildElement("initialStub");

   if(initialStubNode){initialStubIsNull_ = false;}
   else{initialStubIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- initialStubNode , address : " << initialStubNode << std::endl;
   #endif
   if(initialStubNode)
   {
      if(initialStubNode->Attribute("href") || initialStubNode->Attribute("id"))
      {
          if(initialStubNode->Attribute("id"))
          {
             initialStubIDRef_ = initialStubNode->Attribute("id");
             initialStub_ = boost::shared_ptr<Stub>(new Stub(initialStubNode));
             initialStub_->setID(initialStubIDRef_);
             IDManager::instance().setID(initialStubIDRef_,initialStub_);
          }
          else if(initialStubNode->Attribute("href")) { initialStubIDRef_ = initialStubNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { initialStub_ = boost::shared_ptr<Stub>(new Stub(initialStubNode));}
   }

   //finalStubNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* finalStubNode = xmlNode->FirstChildElement("finalStub");

   if(finalStubNode){finalStubIsNull_ = false;}
   else{finalStubIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- finalStubNode , address : " << finalStubNode << std::endl;
   #endif
   if(finalStubNode)
   {
      if(finalStubNode->Attribute("href") || finalStubNode->Attribute("id"))
      {
          if(finalStubNode->Attribute("id"))
          {
             finalStubIDRef_ = finalStubNode->Attribute("id");
             finalStub_ = boost::shared_ptr<Stub>(new Stub(finalStubNode));
             finalStub_->setID(finalStubIDRef_);
             IDManager::instance().setID(finalStubIDRef_,finalStub_);
          }
          else if(finalStubNode->Attribute("href")) { finalStubIDRef_ = finalStubNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { finalStub_ = boost::shared_ptr<Stub>(new Stub(finalStubNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<Stub> StubCalculationPeriod::getInitialStub()
{
   if(!this->initialStubIsNull_){
        if(initialStubIDRef_ != ""){
             return boost::shared_static_cast<Stub>(IDManager::instance().getID(initialStubIDRef_));
        }else{
             return this->initialStub_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<Stub>();
   }
}
boost::shared_ptr<Stub> StubCalculationPeriod::getFinalStub()
{
   if(!this->finalStubIsNull_){
        if(finalStubIDRef_ != ""){
             return boost::shared_static_cast<Stub>(IDManager::instance().getID(finalStubIDRef_));
        }else{
             return this->finalStub_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<Stub>();
   }
}
}

